### PR TITLE
feat: suspend functions as Swift async

### PR DESCRIPTION
**Pull request by p7nov**

### CODE REVIEW

File: docs/topics/native/native-objc-interop.md [hunk 1]
```diff
@@ -133,6 +133,22 @@ Note that the opposite reversed translation is not implemented yet:
 Swift/Objective-C error-throwing methods aren't imported to Kotlin as
 exception-throwing.
 
+### Suspending functions
+
+> The support for calling `suspend` functions from Swift code as `async` is [Experimental](components-stability.md).
+> It may be dropped or changed at any time.
+> Use it only for evaluation purposes. We would appreciate hearing your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issues/KT-47610).
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-objc-interop.md:140]: Wrong URL?

File: docs/topics/native/native-objc-interop.md [hunk 2]
```diff
@@ -133,6 +133,22 @@ Note that the opposite reversed translation is not implemented yet:
...
+>
+{type="warning"}
+
+Kotlin's [suspending functions](coroutines-basics.md) (`suspend`) are presented in the generated Objective-C headers as
+functions with callbacks, or `completionHandler` in the Swift/Objective-C terminology.
+
+Swift 5.5 is bringing the [support for concurrency at the language level](https://docs.swift.org/swift-book/LanguageGuide/Concurrency.html)
+with the `async`/`await` mechanism. Kotlin's `suspend` functions are available for calling from Swift 5.5 and newer as
+`async` functions without using completion handlers. Currently, this functionality is highly experimental and has certain
+limitations. See [this YouTrack](https://docs.swift.org/swift-book/LanguageGuide/Concurrency.html) issue for details.
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-objc-interop.md:150]: Wrong URL?

File: docs/topics/native/native-objc-interop.md [hunk 3]
```diff
@@ -133,6 +133,22 @@ Note that the opposite reversed translation is not implemented yet:
...
+> It may be dropped or changed at any time.
+> Use it only for evaluation purposes. We would appreciate hearing your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issues/KT-47610).
+>
+{type="warning"}
+
+Kotlin's [suspending functions](coroutines-basics.md) (`suspend`) are presented in the generated Objective-C headers as
+functions with callbacks, or `completionHandler` in the Swift/Objective-C terminology.
+
+Swift 5.5 is bringing the [support for concurrency at the language level](https://docs.swift.org/swift-book/LanguageGuide/Concurrency.html)
+with the `async`/`await` mechanism. Kotlin's `suspend` functions are available for calling from Swift 5.5 and newer as
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-objc-interop.md:148]: The documentation doesn't mention that one can call Kotlin `suspend` function as a regular function with completion handler from Swift, in all Swift versions, including 5.5.
Without that, one can read this paragraph wrong.

Comment (p7nov) [reply docs/topics/native/native-objc-interop.md]: Thanks, I rewrote this part for better clarity.

**Review by SvyatoslavScherbina — APPROVED** [review]

**Comment by SvyatoslavScherbina** [comment]
Note: this PR now contains some unrelated commits.

**Review by SvyatoslavScherbina — APPROVED** [review]

**Review by koshachy — APPROVED** [review]
Nice!

**Pull request merged**